_drafts/2021-03-15-data_science_timeseries_plotting_notebook_template.md

```diff
@@ -0,0 +1,146 @@
+[2020_tour]: {% post_url 2020-10-16-2020_tour_de_france_plot %}#the-race-for-yellow
+
+- What does the distribution of these events look like in time? 
+- Are there more of events of Type A or Type B?
+
+Plotting the data as a time series is the best way to answer these questions,
+but I never remember how to pivot the table, aggregate the events by type, and
+resample to the right frequency. So I made the [**Time Series Plotting
+Notebook**][plotting_nb] to remember for me.
+
+[plotting_nb]: https://github.com/agude/Jupyter-Notebook-Template-Library/blob/master/notebooks/basic-time-series-plotting-template.ipynb
+
+## The Time Series Plotting Notebook
+
+Suppose we are interested in the looking at the number of automobile
+collisions by make using my [curated SWITRS dataset][switrs_data]. We could
+extract one row for each collision and the associated vehicle, which would
+look like this:
+
+|  ID  |   datetime |  vehicle_make |
+|:-----|-----------:|--------------:|
+| 0    | 2020-01-01 |         Honda |
+| 1    | 2020-02-01 |        Toyota |
+| 2    | 2020-01-01 |         Other |
+| ...  |        ... |           ... |
+
+[switrs_data]: {% post_url 2020-11-24-switrs_sqlite_hosted_dataset %}
+
+The [**time series plotting notebook**][plotting_nb] has two helpful functions
+to visualize this data.
+
+### Plot Time Series
+
+The first function, `plot_time_series()` is simple. It takes a dataframe
+formatted like the above data and returns a plot showing the number of events
+for each value in the categorical column. For example, to plot the number of
+accidents per week by vehicle make, we would call:
+
+```python
+plot_time_series(
+  df, ax,
+  date_col="datetime",
+  category_col="vehicle_make",
+  resample_frequency="W",  # Resample to 'W'eeks
+)
+```
+
+Which would produce this plot:
+
+[![A simple plot of the number of collisions by vehicle make in
+California][first_plot]][first_plot]
+
+[first_plot]: {{ file_dir }}/make_collision_in_time_first_version.svg
+
+The function has as few optional parameters:
+
+- `resample_frequency` which controls the timescale over which the data is
+aggregated.
+- `aggfunc` which controls how the data is aggregated.
+- `linewidth` which can be used to make the lines larger if there are only a
+few of them, or thinner if there is lots of data.
+
+### Simple Legend
+
+Simple legends are great; they convey their information most effectively
+because all the superfluous noise has been removed. 
+
+My [basic plotting notebook][first_notebook] has a function to remove all the
+extra information from the legend box leaving only the color and the label.
+This time I have taken it a step further: I wrote a function to get rid of the
+box and label each line.
+
+[first_notebook]: {% post_url 2020-07-27-data_science_plotting_notebook_template %}#draw-legends
+
+The function `draw_left_legend()` will draw labels on the end of each line,
```

Comment:
Is it smart about horizontal and/or vertical placement to improve visibility and/or prevent overlap? Brag about this more

Reply:
It's very dumb. :-(